cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/store/MetadataStore.java

```diff
@@ -51,7 +51,7 @@
    * @param propertiesToPreserve the names of properties to leave unchanged even if
    *                             the new metadata contains no or new values for them
    */
-  void replaceMetadata(MetadataScope scope, Metadata metadata,
+  void replaceMetadata(MetadataScope scope, MetadataEntity entity, MetadataDataset.Record metadata,
```

Comment:
javadoc should be updated.

Also, why does this change need to be made if the record contains the entity? What happens if somebody calls this and the entities don't match?

Reply:
Good catch. My intention was to just replace this with Record. This must be a relic of an intermediate refactoring I did. Will remove the entity. By the way, I hope to remove this class entirely in the future as it is merely a proxy translating between old and new metadata APIs. The only thing we need to keep is the audit publishing.